test/testcompute/gaussian_process_test.py

```diff
@@ -99,8 +98,6 @@ def test_constant_liar_update(self, one_hot_domain_list, gaussian_process_list):
       assert gp.num_sampled == num_sampled + 7 + 5 + 3
       assert numpy.all(gp.points_sampled_value[-3:] == mean_value)
 
-
-class TestHasPredictor(GaussianProcessTestCase):
```

Comment:
why are we not testing the `HasPredictor` ? 
I'm fine reconsidering this logic, but we need to thing about it

Reply:
`hpred.predictor` and `gaussian_process` are exactly the same in this instance (`assert hpred.predictor is gaussian_process`). This test class was testing nothing about the implementation of `HasPredictor` except that it has a single attribute `.predictor` which is identical to the `predictor` argument in the constructor. That can be tested as follows:
```python
def test_has_predictor():
  predictor = object()
  hpred = HasPredictor(predictor)
  assert hpred.predictor is predictor
```
IMO this test is not worth keeping. If everything left over here is redundant then it should just be removed.

Reply:
The reason it is relevant in this PR is because `HasPredictor.predictor` is typed as `Predictor`, but this test is testing more than just the predictor interface, it's testing methods that are only defined on `GaussianProcess`.

Reply:
gotcha. I agree that a single test as such:
```
def test_has_predictor():
  predictor = object()
  hpred = HasPredictor(predictor)
  assert hpred.predictor is predictor
```
suffices. Just remove all tests of `TestHasPredictor` then.